app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TweeetsController < ApplicationController
+  get '/tweets/new' do
+    if !session[:user_id]
+      redirect '/login'
+    else
+      erb :'/tweets/create'
+    end
+  end
+  get '/tweets' do
+    @tweets = Tweet.all
+    if !session[:user_id]
+      redirect '/login'
+    else
+      erb :'/tweets/show'
+    end
+  end
+  get '/tweets/:id' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/tweet'
+    else
+      redirect to '/login'
+    end
+  end
+  get '/tweets/:id/edit' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
```

Comment:
I would use the current_user method in the application controller that I had you add here 

```
current_user.id
```